handlers/amazon_s3_file_upload_from_submission_v2/handler/init.rb

```diff
@@ -23,91 +23,255 @@ def initialize(input)
+      #
+      # Process each attachment file
+      field.each_with_index do |attachment_info, index|
+        begin
+          # The attachment file name is stored in the 'name' property
+          attachment_name = attachment_info['name']
+
+          # Temporary file to stream contents to
+          tempdir = "#{Dir.tmpdir}/#{SecureRandom.hex(8)}"
+          tempfile = "#{tempdir}/#{attachment_name}"
+          FileUtils.mkdir_p(tempdir)
+
+
+          # Retrieve the attachment download link from the source server
+          puts "Retrieving attachment download link from source submission: #{attachment_name} for field #{attachment_field}" if @enable_debug_logging
 
-      puts "Submission API route: #{submission_api_route}" if @debug_logging_enabled
+          # API route to get the generated attachment download link from Kinetic Request CE.
+          download_link_api_route = "#{source_server}/app/api/v1" <<
+            "/submissions/#{submission['id']}" <<
+            "/files/#{URI.escape(attachment_field)}" <<
+            "/#{index.to_s}/#{URI.escape(attachment_name)}/url"
 
-      submission = RestClient::Resource.new(submission_api_route,
-        :user => @info_values['request_ce_username'], 
-        :password => @info_values['request_ce_password']).get
-      field = JSON.parse(submission)["submission"]["values"][@parameters["field"]]
+          # Retrieve the URL to download the attachment from Kinetic Request CE.
+          res = http_get(download_link_api_route, {}, source_headers)
+          if !res.kind_of?(Net::HTTPSuccess)
+            message = "Failed to retrieve link for attachment #{attachment_name} from source submission"
+            return handle_exception(message, res)
+          end
+          file_download_url = JSON.parse(res.body)['url']
+          puts "Received link for attachment #{attachment_name} from source submission" if @enable_debug_logging
 
-      return "<results><results name=\"Public Url\"></result></results>" if field.empty?
 
-      # Download the attachment from Request CE/Filehub
-      download_api_route = "#{api_route}/#{URI.escape(@parameters['submission_id'])}/files/"+
-        "#{URI.escape(@parameters['field'])}/0/#{URI.escape(field[0]['name'])}/url"
+          # Download the attachment from the source server
+          puts "Downloading attachment #{attachment_name} from #{file_download_url}" if @enable_debug_logging
+          res = stream_file_download(tempfile, file_download_url, {}, source_headers)
+          if !res.kind_of?(Net::HTTPSuccess)
+            message = "Failed to download attachment #{attachment_name} from the source server"
+            return handle_exception(message, res)
+          end
+          puts "Attachment Name: #{attachment_name}"
+          # Add a path to upload file to if provided
+          destination_upload_path = upload_path.empty? ?
+          attachment_name :
+          "#{upload_path}#{attachment_name}"

```

Comment:
This isn't necessary. It is the same as just doing:

```
destination_upload_path = "#{upload_path}#{attachment_name}"
```

Reply:
If you run the handler in task, I think that leaving upload_path empty with result in nil.  I'll run a test to make sure.

Reply:
I don't think that is right. Nil doesn't have an `.empty?` method, so will raise an error at that point anyway.